src/bootstrap/sanity.rs

```diff
@@ -103,7 +103,9 @@ pub fn check(build: &mut Build) {
         .take()
         .map(|p| cmd_finder.must_have(p))
         .or_else(|| env::var_os("BOOTSTRAP_PYTHON").map(PathBuf::from)) // set by bootstrap.py
-        .or_else(|| Some(cmd_finder.must_have("python")));
+        .or_else(|| cmd_finder.maybe_have("python"))
+        .or_else(|| cmd_finder.maybe_have("python3"))
+        .or_else(|| cmd_finder.maybe_have("python2"));
```

Comment:
Nice, this also matches what `src/tools/x` does :) https://github.com/rust-lang/rust/blob/05d22212e89588e7c443cc6b9bc0e4e02fdfbc8d/src/tools/x/src/main.rs#L28-L43

Reply:
I chose a slightly different order, I figured with letting whatever the system deems the "One True `python`" would be safer to go with first. Seems that one goes `python3` -> `python2` -> `python`, happy to mimic that order.

Reply:
No, your logic matches what it does: https://github.com/rust-lang/rust/pull/95443